beetsplug/vgmdb.py

```diff
@@ -0,0 +1,248 @@
+        self.config.add({
+            'source_weight': 1.0,
+            'lang-priority': 'en, ja-latn, ja',
+            'prefer_original': True  # Only for Attack on Titan soundtracks
+        })
+        log.debug('Querying VGMdb')
+        self.source_weight = self.config['source_weight'].as_number()
+        self.lang = self.config['lang-priority'].get().replace(' ', '').split(",")
+        self.prefer_original = self.config['prefer_original'].get(bool)
+
+    def lang_select(self, d):
+        """Returns the value matching lang-priority or the firt one if not found
+        """
+        if not isinstance(d, dict): return d
+        for lang in self.lang:
+            if lang in d:
+                return d[lang]
+        return list(d.values())[0]
+
+    def lang_select_tracks(self, d):
+        """Same as above for tracks
+        """
+        match = {'en': ['English',
+                        'English (from furigana)',
+                        'English (NA localization + Additional Translations)',
+                        'English (EU localisation + Additional Translations)'],
+                 'ja-latn': ['Romaji'],
+                 'ja': ['Japanese', 'Japanese (furigana)']
+                 }
+
+        if not isinstance(d, dict): return d
+        if self.prefer_original and 'Original' in d: return d['Original']
+        for lang in self.lang:
+            if lang in match:
+                for language in match[lang]:
+                    if language in d:
+                        return d[language]
+        return list(d.values())[0]
+
+    def album_distance(self, items, album_info, mapping):
+        """Returns the album distance.
+        """
+        dist = Distance()
+        if album_info.data_source == 'VGMdb':
+            dist.add('source', self.source_weight)
+        return dist
+
+    def candidates(self, items, artist, album, va_likely, extra_tags=None):
+        """Returns a list of AlbumInfo objects for VGMdb search results
+        matching an album and artist (if not various).
+        """
+        if va_likely:
+            query = album
+        else:
+            query = '%s %s' % (artist, album)
+        try:
+            ret = self.get_albums(query, va_likely)
+            for x in ret:
+                msg = 'Found album: %s ; id: %s ; artist: %s' % (x.album, x.album_id, x.artist)
+                log.debug(msg)
+            return ret
+        except:
+            log.debug('VGMdb Search Error: (query: %s)' % query)
+            return []
+
+    def album_for_id(self, album_id):
+        """Fetches an album by its VGMdb ID and returns an AlbumInfo object
+        or None if the album is not found.
+        """
+        log.debug('Querying VGMdb for release %s' % str(album_id))
+
+        # Get from VGMdb
+        r = requests.get('http://vgmdb.info/album/%s?format=json' % str(album_id))
+
+        # Decode Response's content
+        try:
+            item = r.json()
+        except:
```

Comment:
Plain `except:` blocks are dangerous—perhaps handle a `JSONDecodeError` or whatever it's called?